opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -103,7 +105,7 @@ impl LogProcessor for SimpleLogProcessor {
 
     #[cfg(feature = "logs_level_enabled")]
     fn event_enabled(&self, _level: Severity, _target: &str, _name: &str) -> bool {
-        true
+        !Context::current().suppress_logs
```

Comment:
we should not tie this to the `logs_level_enabled` feature.

Reply:
Thanks for the comment. I believe placing it within `logs_level_enabled` is preferable for early suppression detection, which would avoid the need to generate a `LogRecord`. I also plan to add this validation in `Logger::emit()` for instances when the `logs_level_enabled` feature is not enabled, will be doing it before making it ready for review.

Reply:
Agree doing it at the earliest possible place is perf friendly!

Can we do `fn event_enabled` always (i.e even without logs_level_enabled feature) and check Suppression inside that? And if logs_level_enabled feature is enabled, then do the additional check?

Reply:
Yes we can do that, adding this validation within appender for logs and tracing, or better in LogRecord->Emit() or similar method once have implemented #1189. As of now, I am still bit struggling with propagating the future state across threads used by async task, so will revisit that once it is covered.